artiq/coredevice/rtio_batch.py

```diff
@@ -0,0 +1,37 @@
+from artiq.language.core import syscall, kernel
+from artiq.language.types import TInt32, TInt64, TStr, TNone, TTuple, TBool
+
+from numpy import int64
+
+@syscall
+def rtio_batch_start() -> TNone:
+    raise NotImplementedError("syscall not simulated")
+
+@syscall
+def rtio_batch_end() -> TNone:
+    raise NotImplementedError("syscall not simulated")
+
+
+class RTIOBatch:
+    """Context manager for batching RTIO events.
+
+    All output RTIO events within the context will be buffered
+    on the core device and executed immediately after leaving the context.
+
+    This feature is available only on Zynq devices such as Kasli-SoC,
+    ZC706 and EBAZ4205 with ACP Kernel Initiator enabled.
+
+    Unlike DMA, there is no effect on the value of ``now``.
+    """
+    def __init__(self, dmgr, core_device="core"):
+        # since this is just a syscall wrapper for semantics,
+        # nothing has to be done in init
+        pass
+
+    @kernel
+    def __enter__(self):
+        rtio_batch_start()
+
+    @kernel
+    def __exit__(self, type, value, traceback):
+        rtio_batch_end()
```

Comment:
I would just add this to ``artiq.coredevice.rtio`` instead of creating another short module.